test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -184,6 +184,43 @@ def test_tf_serving_version_cpu(tensorflow_inference):
 
     stop_and_remove_container(container_name, ctx)
 
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.model("N/A")
+def test_tf_serving_api_version(tensorflow_inference):
+    """
+    For non-huggingface TF inference images, check that the tag version matches the version of TF serving api
+    in the container.
+
+    Huggingface includes MMS and core TF, hence the versioning scheme is based off of the underlying tensorflow
+    framework version, rather than the TF serving version.
+
+    @param tensorflow_inference: ECR image URI
+    """
+    # Set local variable to clarify contents of fixture
+    image = tensorflow_inference
+
+    if "gpu" in image:
+        cmd="pip show tensorflow-serving-api-gpu | grep Version"
+    if "cpu" in image:
+        cmd="pip show tensorflow-serving-api | grep Version"
+
+    _, tag_framework_version = get_framework_and_version_from_tag(image)
+    
+    ctx = Context()
+    container_name = get_container_name("tf-serving-api-version", image)
+    start_container(container_name, image, ctx)
+    try:
+        output = run_cmd_on_container(
+            container_name, ctx, cmd, executable="bash"
+        )
+        str_version_from_output = ((str(output.stdout).split(' '))[1]).strip()
+        assert (tag_framework_version == str_version_from_output), \
+            f"Tensorflow serving API version is {str_version_from_output} while the Tensorflow version is {tag_framework_version}. Both don't match!"
+    except Exception as e:
+        LOGGER.error(f"Unable to execute command on container. Error: {e}")
+        raise        
+    finally:
```

Comment:
Let's remove finally here except if start_container is going to be moved into the try-except block. Also, verify what happens if we cannot start a container with the name. Does the command "start_container" return a value? We can use that value to decide if we need to raise an exception or stop the container after running the test command. 

The assumptions in this piece of code are 1. container name is always given and the containers are always instantiated successfully.

Reply:
The container doesnt need to be inside try block as per the review. The start container can fail by itself and that has nothing to do with finally. The stop and remove container logic only executes when something in the try block fails because we know that the code within try gets executed only upon successful start of container and we want a cleanup/removal of such.